examples/mnit_moles/src/moles.nit

```diff
@@ -23,64 +23,81 @@ module moles is
 end
 
 import mnit
+import realtime
 
+import drawing
+
+# A hole with a possible mole, or a trap in it
 class Hole
 	var game: Game
 
-	# Center of the hole
+	# Horizontal center of the hole
 	var x: Int
+
+	# Vertical center of the hole
 	var y: Int
 
 	# Half width of the hit box
 	var dx = 200.0
-	# Heigth of the hit box
+
+	# Height of the hit box
 	var dy = 800.0
 
-	# state
-	var up = false
-	var hitted = false
-	var trap = false
+	# Content (and state) of this hole
+	var content: nullable HoleContent = null
 
-	init (g: Game, x, y: Int)
-	do
-		game = g
-		self.x = x
-		self.y = y
+	# Time at witch this hole can act again
+	var timeout_until: Float = app.clock.total.to_f is lazy
+
+	# Timeout for 1/2 sec or less with a higher score
+	private fun reset_timeout do
+		timeout_until = app.clock.total.to_f +
+		                0.5 - 0.4 * (game.points.to_f / 1000.0).min(1.0)
 	end
 
 	fun do_turn
 	do
-		if up then
-			if hitted then
+		if timeout_until > app.clock.total.to_f then return
+
+		var content = content
+		if content != null then
+			if content == game.down then
 				if (20.0*game.speed_modifier).to_i.rand == 0 then
 					# dead / hide
-					hitted = false
-					up = false
+					self.content = null
+					reset_timeout
 				end
 			else if (80.0*game.speed_modifier).to_i.rand == 0 then
 				# hide
-				up = false
+				self.content = null
+				reset_timeout
 			end
 		else if (100.0*game.speed_modifier).to_i.rand == 0 then
-			# show up
-			up = true
+			self.content = to_pop
+			reset_timeout
+		end
+	end
 
-			# shot traps only at 50 points and up
-			trap = false
-			if game.points > 50 then
+	# Get next `HoleContent` to pop
```

Comment:
> It's in the doc... the content to_pop.

Yes, I read doc. I read your last comment. And I still have no clue of what `pop` means in this context. 

Also it's in the `examples/` directory rigth? So it's have to be taken very seriously. For butched things, there is `contrib/`.